src/storage/src/hummock/state_store.rs

```diff
@@ -95,65 +93,61 @@ impl HummockStorage {
-                        prune_ssts(level.get_table_infos().iter(), &key_range, backward, None);
-                    if table_infos.is_empty() {
-                        continue;
-                    }
-
-                    for table_info in table_infos.into_iter().rev() {
-                        let table = self.sstable_store.sstable(table_info.id).await?;
-                        if backward {
-                            overlapped_backward_iters.push(Box::new(BackwardSSTableIterator::new(
-                                table,
-                                self.sstable_store(),
-                            ))
-                                as BoxedBackwardHummockIterator);
-                        } else {
-                            overlapped_forward_iters.push(Box::new(SSTableIterator::new(
-                                table,
-                                self.sstable_store(),
-                            ))
-                                as BoxedForwardHummockIterator);
-                        };
-                    }
-                }
-                LevelType::Nonoverlapping => {
-                    if level.get_table_infos().is_empty() {
-                        continue;
-                    }
-
-                    let start_table_idx = match key_range.start_bound() {
-                        Included(key) | Excluded(key) => search_sst_idx(level, key),
-                        _ => 0,
-                    };
-                    let end_table_idx = match key_range.end_bound() {
-                        Included(key) | Excluded(key) => search_sst_idx(level, key),
-                        _ => level.table_infos.len().saturating_sub(1),
-                    };
-                    assert!(
-                        start_table_idx < level.table_infos.len()
-                            && end_table_idx < level.table_infos.len()
-                    );
-                    let table_infos = &level.get_table_infos()[start_table_idx..=end_table_idx];
+            let table_infos =
+                prune_ssts(level.get_table_infos().iter(), &key_range, backward, None);
+            if table_infos.is_empty() {
+                continue;
+            }
 
-                    if backward {
-                        overlapped_backward_iters.push(Box::new(BackwardConcatIterator::new(
-                            table_infos.iter().rev().cloned().collect(),
-                            self.sstable_store(),
-                        ))
-                            as BoxedBackwardHummockIterator);
-                    } else {
-                        overlapped_forward_iters.push(Box::new(ConcatIterator::new(
-                            table_infos.iter().cloned().collect_vec(),
-                            self.sstable_store(),
-                        ))
-                            as BoxedForwardHummockIterator);
-                    };
-                }
+            for table_info in table_infos.into_iter().rev() {
```

Comment:
SSTs in L1 overlap with one another after this change and we haven't implemented the logic to pass the vnode mapping to state store read interfaces. In this case, we will have a very lage merge iterator and I think this is why e2e times out.

Reply:
> SSTs in L1 overlap with one another after this change and we haven't implemented the logic to pass the vnode mapping to state store read interfaces. In this case, we will have a very lage merge iterator and I think this is why e2e times out.

No. When I treated L1 as non-overlapping (i.e. not to ensure correctness), it still went time out